astropy/io/misc/pandas/connect.py

```diff
@@ -97,6 +98,15 @@ def _pandas_write(fmt, tbl, filespec, **kwargs):
     df = tbl.to_pandas()
     write_method = getattr(df, 'to_' + pandas_fmt)
 
+    if overwrite is not True:
+        try:  # filespec is not always a path-like
+            exists = os.path.exists(filespec)
```

Comment:
I noticed in https://docs.python.org/3/library/os.path.html#os.path.exists that `filespec` can be an integer and `os.path.exists` will return True/False if the file descriptor for that number exists. Not clear if that is a problem here..

I was just wondering about this in the context of the `TypeError` below, which I don't know if that is the *only* exception that `exists` can raise. The intent of this code block is never to raise an exception apart from `OSError` if the filespec is PathLike and it exists.

Anyway, most likely this is OK, just wondering.

Reply:
Good notice. I'll dig a little deeper on this.

Reply:
Ideally,  `overwrite` would be interpreted only to the extent that it affects how a file is opened:  `mode='x'` for `overwrite=False` and `mode='w'` for `overwrite=True`; if the user passes in anything that does not need to be opened, they are on their own...

For a moment, I thought things were easy, since `DataFrame.to_csv` takes a `mode` argument, but that does not seem to be shared with the other formats.

Anyway, overall probably easiest just to stick with what you have here - it captures the main use cases.

Aside: I would just to `if overwrite:` - don't see the need to disallow `np.True_`...

Reply:
Done!